src/api/app/models/status_message.rb

```diff
@@ -29,6 +29,16 @@ def delete
 
   def acknowledge!
     users << User.session!
```

Comment:
```suggestion
unless users.include?(User.session!)
    users << User.session!
```

Reply:
Then you dont need the `elsif` in the controller

Reply:
Yeah I already had this in consideration when thinking on the solution. The point is the code in the suggestion is checking the association against the database every time the user clicks on the `Got it` button. The code in the PR will only make the check against the association when we're not in the happy path.

But thanks anyway @krauselukas !

Reply:
I don't get it @danidoni , the user should click just once on the "Got it", so I don't see the issue with @krauselukas suggestion

Reply:
Mmm you are both right, I got confused somehow. I'm addressing it now.

Reply:
Check it out again, please!